### ISSUE TITLE
Potential bug in PPOTrainer code. 

### ISSUE THREAD

**Issue by sverneka** [lvwerra/trl#334]
Hi, I am comparing your implementation to that of OpenAI's implementation (https://github.com/openai/lm-human-preferences/blob/cbfd210bb8b08f6bc5c26878c10984b90f516c66/lm_human_preferences/train_policy.py#L285)

The difference is that PPO is trained only on the response tokens in OpenAI's implementation vs in your case, it is trained on query+response. Is that a bug or just another variant?

Thanks
Sachin

**Comment by sverneka** [comment]
@younesbelkada As far as logprobs and ref_logprobs are concerned that are used by the PPOTrainer, it only gets it for response and not for query+response. 

https://github.com/openai/lm-human-preferences/blob/cbfd210bb8b08f6bc5c26878c10984b90f516c66/lm_human_preferences/policy.py#L114
In this code, ref_logprobs is obtained as below, here you can see they have : logits = result['logits'][:, context_length-1:-1] where context_length is length of query.

    def analyze_responses_op(self, queries, responses):
        contexts = self.embed_queries(queries)
        context_length = tf.shape(contexts)[1]
        tokens = tf.concat([contexts, responses], axis=1)
        result = self.step_core(self.model_hparams, tokens)
        logits = result['logits'][:, context_length-1:-1]

        logits /= self.temperature
        return dict(
            logprobs = utils.logprobs_from_logits(logits=logits, labels=responses),
            entropies = utils.entropy_from_logits(logits),
            values = result['values'][:, context_length-1:-1],
        )
